src/audio/volume/volume.c

```diff
@@ -413,6 +422,8 @@ static struct comp_dev *volume_new(const struct comp_driver *drv,
 	dev->state = COMP_STATE_READY;
 	return dev;
 
+vol_fail:
+	rfree(cd->vol);
```

Comment:
instead of adding another label you can just put this under cd_fail since rfree(NULL) is a no-op